src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    dateMap[fromFormat[i]] = dateParts[i];
+  }
+
+  const year =
+    dateMap['YYYY'] ||
+    (dateMap['YY']
+      ? (parseInt(dateMap['YY'], 10) < 30 ? '20' : '19') + dateMap['YY']
+      : null);
+  const month = dateMap['MM'];
+  const day = dateMap['DD'];
+
+  const outputParts = toFormat.map((formatPart) => {
+    if (formatPart === 'YYYY') {
+      return year;
+    }
+
+    if (formatPart === 'YY') {
+      return year.slice(-2);
+    }
+
+    if (formatPart === 'MM') {
+      return month;
+    }
+
+    if (formatPart === 'DD') {
+      return day;
+    }
+
+    return '';
```

Comment:
The empty string return for a non-matching format part is unnecessary. Since the `toFormat` array will only contain valid date parts and the separator, you can assume that every `formatPart` will match one of the conditions. Thus, the default return case can be omitted.